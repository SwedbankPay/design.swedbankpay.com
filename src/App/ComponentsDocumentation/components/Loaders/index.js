import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import CodeTags from "@components/CodeTags";

const BasicUsage = () => (
    <>
        <h2 id="basic-usage">Basic usage</h2>
        <p>
            To use a default loader add <CodeTags type="secondary" code=".loader" /> to a <CodeTags type="primary" code="<div>" />.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="loader" />
        </ComponentPreview>
    </>
);

const CenteredLoader = () => (
    <>
        <h2 id="centered-loader">Centered loader</h2>
        <p>
            To center the loader add class <CodeTags type="secondary" code=".loader-center" />.
            <ComponentPreview language="html" showCasePanel codeFigure>
                <div className="loader loader-center" />
            </ComponentPreview>
        </p>
    </>
);

const Sizes = () => (
    <>
        <h2 id="sizes">Sizes</h2>
        <p>In addition to <CodeTags type="secondary" code=".loader" /> you can use <CodeTags type="secondary" code=".loader-sm" /> and <CodeTags type="secondary" code=".loader-lg" /> to decrease or increase the size of the loader.</p>
        <h3>Small</h3>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="loader loader-sm" />
        </ComponentPreview>
        <h3>Large</h3>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="loader loader-lg" />
        </ComponentPreview>
    </>
);

const MutedLoader = () => (
    <>
        <h2 id="muted-loader">Muted loader</h2>
        <p>
            Use class <CodeTags type="secondary" code=".loader-muted" /> to make the loader muted.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="loader loader-muted" />
        </ComponentPreview>
    </>
);

const Loaders = () => (
    <DocContainer docToc>
        <p className="lead">
            Use loaders to indicate to the user that your web application is loading.
        </p>
        <BasicUsage />
        <CenteredLoader />
        <Sizes />
        <MutedLoader />
    </DocContainer>
);

export default Loaders;

/* For testing */
export { BasicUsage, CenteredLoader, Sizes, MutedLoader };
