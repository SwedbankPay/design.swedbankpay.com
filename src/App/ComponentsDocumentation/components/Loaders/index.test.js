import React from "react";
import { shallow } from "enzyme";

import Loaders, { BasicUsage, CenteredLoader, Sizes, MutedLoader } from "./index";

describe("Documentation: Loaders", () => {
    it("is defined", () => {
        expect(Loaders).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Loaders />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Basic Usage", () => {
        it("is defined", () => {
            expect(BasicUsage).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<BasicUsage />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CenteredLoader", () => {
        it("is defined", () => {
            expect(CenteredLoader).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CenteredLoader />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Sizes", () => {
        it("is defined", () => {
            expect(Sizes).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Sizes />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Muted Loader", () => {
        it("is defined", () => {
            expect(MutedLoader).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<MutedLoader />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
