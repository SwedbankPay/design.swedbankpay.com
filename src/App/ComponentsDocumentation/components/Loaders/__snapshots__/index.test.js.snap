// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: Loaders Basic Usage renders 1`] = `
<Fragment>
  <h2
    id="basic-usage"
  >
    Basic usage
  </h2>
  <p>
    To use a default loader add 
    <CodeTags
      code=".loader"
      type="secondary"
    />
     to a 
    <CodeTags
      code="<div>"
      type="primary"
    />
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="loader"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Loaders CenteredLoader renders 1`] = `
<Fragment>
  <h2
    id="centered-loader"
  >
    Centered loader
  </h2>
  <p>
    To center the loader add class 
    <CodeTags
      code=".loader-center"
      type="secondary"
    />
    .
    <ComponentPreview
      codeFigure={true}
      language="html"
      showCasePanel={true}
    >
      <div
        className="loader loader-center"
      />
    </ComponentPreview>
  </p>
</Fragment>
`;

exports[`Documentation: Loaders Muted Loader renders 1`] = `
<Fragment>
  <h2
    id="muted-loader"
  >
    Muted loader
  </h2>
  <p>
    Use class 
    <CodeTags
      code=".loader-muted"
      type="secondary"
    />
     to make the loader muted.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="loader loader-muted"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Loaders Sizes renders 1`] = `
<Fragment>
  <h2
    id="sizes"
  >
    Sizes
  </h2>
  <p>
    In addition to 
    <CodeTags
      code=".loader"
      type="secondary"
    />
     you can use 
    <CodeTags
      code=".loader-sm"
      type="secondary"
    />
     and 
    <CodeTags
      code=".loader-lg"
      type="secondary"
    />
     to decrease or increase the size of the loader.
  </p>
  <h3>
    Small
  </h3>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="loader loader-sm"
    />
  </ComponentPreview>
  <h3>
    Large
  </h3>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="loader loader-lg"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Loaders renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Use loaders to indicate to the user that your web application is loading.
  </p>
  <BasicUsage />
  <CenteredLoader />
  <Sizes />
  <MutedLoader />
</DocContainer>
`;
